spec/event_tracer_spec.rb

```diff
@@ -98,8 +85,36 @@
         end
       end
 
-      context 'Logger fails gracefully when exception occurs' do
-        it_behaves_like 'error_in_logger_service_fails_gracefully'
+      context 'Logger fails when error occurs' do
+        before do
+          expect(mock_logger).to receive(selected_log_method).and_raise(RuntimeError.new('Runtime error in base logger'))
+        end
+
+        it 'raises the original error' do
+          expect { subject.public_send(selected_log_method, **args) }.to raise_error do |error|
+            expect(error).to be_a(RuntimeError)
+            expect(error.message).to eq('Runtime error in base logger')
+          end
+        end
+
+        context 'when there is a configured error handler' do
+          before do
+            EventTracer::Config.config.error_handler = ->(error, payload) { puts error, payload }
+          end
+
+          after do
+            EventTracer::Config.reset_config
```

Comment:
This can clear the config defined in `spec_helper.rb`. Is this the behavior we want?

Reply:
Good point! Have moved the config setting to the specific spec that requires it.